accepted/2022/simplify-output-paths.md

```diff
@@ -0,0 +1,53 @@
+# Simplify .NET Output Paths
+
+We'd like to simplify the output paths for .NET 8.
+
+Currently, the default output path for a .NET project includes folders for the `TargetFramework` and the `RuntimeIdentifier` (if it is set).  Additionally, the publish output goes in a `publish` subfolder of the output folder.  The resulting paths are as follows:
+
+- `bin\<Configuration>\<TargetFramework>` - Build output with no `RuntimeIdentifier`
+- `bin\<Configuration>\<TargetFramework>\<RuntimeIdentifier>` - Build output with `RuntimeIdentifier`
+- `bin\<Configuration>\<TargetFramework>\publish` - Publish output with no `RuntimeIdentifier`
+- `bin\<Configuration>\<TargetFramework>\<RuntimeIdentifier>\publish` - Publish output with `RuntimeIdentifier`
+
+This makes it harder to find the output of a project, and sometimes nests one type of output in another output folder.
+
+## Proposed behavior
+
+Projects targeting .NET 8 and higher will by default not include the `TargetFramework` in the output path unless the project is multi-targeted.  Similarly, the `RuntimeIdentifier` will not by default be part of the output path.  Finally, the publish output path will be `bin\publish` by default instead of a `publish` subfolder inside of the output path.
```

Comment:
> Similarly, the `RuntimeIdentifier` will not by default be part of the output path.

Would it make sense to skip the `RuntimeIdentifier` only if the project is published for native SDK RID (ie only if the project is not cross-compiled)?

Reply:
Perhaps, but I think we should try to avoid having one publish folder inside of another.  I think in many cases you're going to be copying the entire publish folder, so if the cross-compiled publish outputs are in subfolders of the native RID publish folder, then when you go to copy that folder you'll also accidentally get all of the other RIDs you published for.

I think maybe it's better for projects that want to go back and forth between publishing different RIDs to set the `AppendRuntimeIdentifierToOutputPath` property to true.  It [sounds like](https://github.com/dotnet/designs/pull/278#pullrequestreview-1152768485) the iOS SDK would probably want to do this by default, for example.